game.js

```diff
@@ -0,0 +1,148 @@
+let bot = require("./bot.js");
+let helpers = require("./helpers.js");
+let lastLetter = "";
+let cities = [
+  "Архангельск",
+  "Борисполь",
+  "Воронеж",
+  "Гюмри",
+  "Донецк",
+  "Енакиево",
+  "Жуковский",
+  "Запорожье",
+  "Изюм",
+  "Йошкар-Ола",
+  "Киев",
+  "Лондон",
+  "Москва",
+  "Николаев",
+  "Орёл",
+  "Париж",
+  "Ростов",
+  "Селидово",
+  "Трускавец",
+  "Уфа",
+  "Флоренция",
+  "Харьков",
+  "Цюрих",
+  "Челябинск",
+  "Шанхай",
+  "Щёлково",
+  "Ыгдыр",
+  "Электросталь",
+  "Юрмала",
+  "Ялта"
+].map(item => item.toLowerCase());
+
+exports.session = class Session {
+  constructor(id, chatId = 0) {
+    this.id = id;
+    this.chatId = chatId;
+    this.spentCities = new Set();
+  }
+};
+
+function randomCity(arrCities) {
+  return arrCities[helpers.getRandomInt(arrCities.length)];
+}
+
+function lastValidLetter(str) {
+  let lastLetter = str[str.length - 1];
+  let invalidLetters = ["ь", "Ъ"];
+  let i = 2;
+
+  while (invalidLetters.includes(lastLetter) || str.length == i) {
+    lastLetter = str[str.length - i];
+    i++;
+  }
+  return lastLetter;
+}
+
+function checkCityInDB(sessionId, city, letter) {
+  let result = { isError: false, errorMsg: "" };
+
+  if (letter != city[0]) {
+    result.errorMsg = "Нужно назвать город на букву " + letter.toUpperCase();
+    result.isError = true;
+    return result;
+  }
+  if (bot.session[sessionId].spentCities.has(city)) {
+    result.errorMsg = "Этот город уже был назван!";
+    result.isError = true;
+  }
+  return result;
+}
+
+function selectCityByLetter(sessionId, letter, cities) {
+  let findCities = [];
+  findCities = cities.filter(
```

Comment:
Why do you need `.filter` if you are looking for 1 city. Use `find` instead

Reply:
filter allows to find in array of cities all the cities who named on selected letter, and sselect only cities who were not called. And then i select random city from filtered array. It will give dynamics to the game.